pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -110,12 +114,10 @@ func NewController(
 			UpdateFunc: controller.PassNew(impl.EnqueueControllerOf),
 		},
 	})
-	virtualServiceInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Route")),
-		Handler: cache.ResourceEventHandlerFuncs{
-			AddFunc:    impl.EnqueueControllerOf,
-			UpdateFunc: controller.PassNew(impl.EnqueueControllerOf),
-		},
+
+	clusterIngressInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.enqueueOwnerRoute(impl),
+		UpdateFunc: controller.PassNew(c.enqueueOwnerRoute(impl)),
```

Comment:
I'm surprised at the fact that the controllers (not only networking related controllers, but all of the existing controllers) do not watch deletion events from child resources. Any special reasons that force us to make it this way?

I think it fairly reasonable and necessary for a controller to watch the events, and create one if its resource got deleted mistakenly.